### ISSUE TITLE
no-unpublished-require ignores package.json's main declaration

### ISSUE THREAD

**Issue by reconbot** [mysticatea/eslint-plugin-node#86]
I have some requires that specify the package root (eg `require('../')`) and not the package main file (`../lib/index.js`) this works fine for requires. These are marked as invalid when they're not.

**Comment by mysticatea** [comment]
Thank you for this report.

But I'm sorry, I have not understood what you said.
Please tell me more detail:

- What did you do? (repro steps)
- What did you expect?
- What did it happen?

**Comment by reconbot** [comment]
I have a package with the files.

```
lib/index.js
bin/foo.js
package.json
```

package.json
```
{
  main: 'lib'
}
```

bin/foo.js
```
require('../') // <--- eslint error
```

I expected `bin/foo.js` to not have an error, it instead triggers `node/no-missing-require` and `node/no-unpublished-require`.

**Comment by mysticatea** [comment]
Thank you for the clarification.

Could you confirm this issue on the latest version (v5.1.0)?
In fact, it changed the way to resolve paths in v5.1.0.

**Comment by reconbot** [comment]
I can confirm this happens on 5.1.0

**Comment by reconbot** [comment]
I believe this is actually fixed, my `package-lock.json` had some older versions of dependencies.
